kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -145,6 +148,64 @@ public static Optional<Metadata> assignMaterializedColumnNamesIfNeeded(Metadata
         : Optional.of(metadata.withMergedConfiguration(configsToAdd));
   }
 
+  /**
+   * Converts a logical row tracking field to its physical counterpart.
+   *
+   * <p>This method does not modify the logical schema but returns a new StructType instance with
+   * the added physical field(s).
+   *
+   * <p>Since computing the row ID requires the row index, requesting a row tracking column can
+   * require adding two columns to the physical schema.
+   *
+   * <p>Note that we must not mark the physical columns as metadata columns because as far as the
+   * parquet reader is concerned, these columns are not metadata columns.
+   *
+   * @param logicalField The logical field to convert.
+   * @param logicalSchema The logical schema containing the field.
+   * @param physicalSchema The current physical schema to which the field will be added.
+   * @param metadata The current metadata of the table.
+   * @return A new StructType representing the updated physical schema with the added field.
+   */
+  public static StructType convertToPhysicalColumn(
```

Comment:
Is there a reason this can't return a StructField instead of a StructType? (and add the StructField to the schema outside this fx?) it sits strangely in the other function this way

Reply:
Row IDs are generally not materialized unless they are preserved during update/merge operations (kernel does not preserve IDs at the moment but the spec encourages it). To compute non-materialized row IDs, we need the row index.